python/tvm/tir/buffer.py

```diff
@@ -187,13 +187,11 @@ def __getitem__(self, indices):
         if any(isinstance(index, slice) and index.step is None for index in indices):
             region = []
             analyzer = Analyzer()
-            for index in indices:
+            for i, index in zip(range(len(indices)), indices):
                 if isinstance(index, slice):
-                    region.append(
-                        Range.from_min_extent(
-                            index.start, analyzer.simplify(index.stop - index.start)
-                        )
-                    )
+                    start = 0 if index.start is None else index.start
+                    stop = self.shape[i] if index.stop is None else index.stop
+                    region.append(Range.from_min_extent(start, analyzer.simplify(stop - start)))
```

Comment:
Probably we should check if `index.step` is 1 here to ensure safety.

Reply:
yep we don't allow non-trivial `index.step` here